presto-spark-base/src/test/java/com/facebook/presto/spark/execution/TestNativeExecutionProcess.java

```diff
@@ -71,6 +77,18 @@ public void testNativeProcessRelaunch()
         assertNotSame(process2, process);
     }
 
+    @Test
+    public void testNativeProcessShutdown()
+    {
+        Session session = testSessionBuilder().setSystemProperty(NATIVE_EXECUTION_EXECUTABLE_PATH, "/bin/echo").build();
+        NativeExecutionProcessFactory factory = createNativeExecutionProcessFactory();
+        // Set the maxRetryDuration to 0 ms to allow the RequestErrorTracker failing immediately
+        NativeExecutionProcess process = factory.createNativeExecutionProcess(session, BASE_URI, new Duration(0, TimeUnit.MILLISECONDS));
+        Throwable exception = expectThrows(PrestoSparkFatalException.class, process::start);
+        assertTrue(exception.getMessage().contains("Native process launch failed with multiple retries"));
+        assertFalse(process.isAlive());
```

Comment:
I wonder where is the code that kills the process.

Reply:
It's  L145 in [NativeExecutionProcess.java](https://github.com/prestodb/presto/pull/19842/files#diff-37dd863f0b5ab2886e298455ef640091a9053642f992edd31e5c9ba59c968134) right before we throw the `PrestoSparkFatalException`